### PR TITLE
PUB-1149 Add Authentication

### PR DESCRIPTION
**Pull request by chris-pateman**
### JIRA link (if applicable) ###
PUB-1149

### Change description ###
Add teh authentication for the APIs

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x ] No
```

### CODE REVIEW

File: infrastructure/data.tf [hunk 1]
```diff
@@ -27,16 +27,18 @@ data "azurerm_api_management_product" "apim_product" {
   api_management_name = local.apim_name
 }
 
-/* data "azurerm_key_vault" "b2c" {
-  name                = local.key_vault_name
-  resource_group_name = "${var.product}-ss-${local.env}-rg"
+data "azurerm_key_vault_secret" "data_client_pwd" {
```
Comment (lowflying) [inline infrastructure/data.tf:30]: what happend to the for_each love!?

**Review by lowflying — APPROVED** [review]
Mostly joking cuz they're only kv secrets ;)

**Review by adamtodd0 — APPROVED** [review]
Looks good

File: infrastructure/data.tf (hunk 1 above)
Comment (chris-pateman) [reply infrastructure/data.tf:30]: does like being a conditional haha

**Pull request merged**